feature_engine/timeseries/forecasting/time_lag.py

```diff
@@ -0,0 +1,173 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Union
+
+import pandas as pd
+from pandas.tseries.offsets import DateOffset
+from pandas.tseries.frequencies import to_offset
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from datetime import datetime
+
+from feature_engine.dataframe_checks import (
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import (
+    _find_all_variables,
+    _check_input_parameter_variables,
+)
+
+
+class TimeSeriesLagTransformer(BaseEstimator, TransformerMixin):
+    """
+
+    Parameters
+    ----------
+    periods: int, default=1
+        Number of periods to shift. Can be positive or negative.
+
+    freq: str, default=None
+        Optional increment to use from the tseries module or time rule.
+
+    axis: int, default=1
+        Shift direction. Index is '0'. Columns are '1'.
+
+    fill_value: [int, float, str], default=None
+        Character to be used to fill missing values.
+
+    keep_original: bool, default=True
+        Determines whether the dataframe keeps the original columns
+        that are transformed.
+
+    Attributes
+    ----------
+
+
+    Methods
+    -------
+
+
+    Notes
+    -----
+
+
+    See Also
+    --------
+
+
+    References
+    ----------
+
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        periods: int = 1,
+        freq: str = None,
+        axis: int = 0,
+        fill_value: [int, str] = None,
+        keep_original: bool = True,
+
+    ) -> None:
+
+        if not isinstance(periods, int):
+            raise ValueError(
+                f"'num_periods' is {periods}. The variable must be "
+                f"an integer."
+            )
+
+        # add check for 'freq' param
+
+        if axis not in (0, 1):
+            raise ValueError(
+                f"'axis' is {axis}. The variable must be 0 or 1."
+            )
+
+        if not isinstance(keep_original, bool):
+            raise ValueError(
+                f"'keep_original' is {keep_original}. The variable must "
+                f"be boolean."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.periods = periods
+        self.freq = freq
+        self.axis = axis
+        self.fill_value = fill_value
+        self.keep_original = keep_original
+
```

Comment:
all our classes need a fit method. We should check that X is a df, and that the user either indicated numerical variables, or automatically select numerical variables. And a couple of more attributes from fit like n_features_in

Reply:
Should the transformer only transform numerical variables? Do we keep the categorical variables constant?

Reply:
the transformer ignores categorical variables. If user passes one, it would raise an error, which is already part of _find_or_check_numerical_variables functionality.